Typescript type definition complains about 'function'

When I run cypress with typescript and cypress-map, I got  the following error from typescript:

```
node_modules/cypress-map/commands/index.d.ts:113:29 - error TS2552: Cannot find name 'function'. Did you mean 'Function'?

113     print(format?: string | function): Chainable<any>
                                ~~~~~~~~

  node_modules/typescript/lib/lib.es5.d.ts:318:13
    318 declare var Function: FunctionConstructor;
                    ~~~~~~~~
    'Function' is declared here.
```
The script still works.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -7,13 +7,24 @@ jobs:
       - name: Checkout 🛎
         uses: actions/checkout@v3
 
-      - name: Run Cypress tests 🧪
+      - name: Install everything 📦
         # https://github.com/cypress-io/github-action
         uses: cypress-io/github-action@v5
         with:
-          # make sure we did not leave "it.only" accidentally
-          # https://github.com/bahmutov/stop-only
-          build: npm run stop-only
+          runTests: false
+
+      # make sure we did not leave "it.only" accidentally
+      # https://github.com/bahmutov/stop-only
+      - name: Catch "it.only" 🫴
+        run: npm run stop-only
+
+      - name: Lint code ☑️
+        run: npm run lint
+
+      - name: Run Cypress tests 🧪
+        uses: cypress-io/github-action@v5
+        with:
+          install: false
 
   release:
     needs: [test]

--- a/commands/index.d.ts
+++ b/commands/index.d.ts
@@ -67,7 +67,10 @@ declare namespace Cypress {
      * @example
      *  cy.wrap(2).partial(Cypress._.add, 4).should('equal', 6)
      */
-    partial(fn: Function, ...knownArguments: unknown[]): Chainable<any>
+    partial(
+      fn: Function,
+      ...knownArguments: unknown[]
+    ): Chainable<any>
 
     /**
      * A query command that returns the first element / item from the subject.
@@ -97,7 +100,7 @@ declare namespace Cypress {
      * @example
      *  cy.wrap({ age: '20' }).update('age', Number).should('deep.equal', {age: 20})
      */
-    update(prop: string, callback: function): Chainable<any>
+    update(prop: string, callback: Function): Chainable<any>
 
     /**
      * Returns an object or DOM element from the collection at index K.
@@ -131,7 +134,7 @@ declare namespace Cypress {
      * @example
      *  cy.wrap(arr).print('array length {0.length}')
      */
-    print(format?: string | function): Chainable<any>
+    print(format?: string | Function): Chainable<any>
 
     /**
      * Collects all cells from the table subject into a 2D array of strings.
@@ -140,7 +143,12 @@ declare namespace Cypress {
      * @example cy.get('table').table()
      * @example cy.get('table').table(0, 0, 2, 2)
      */
-    table(x?: number, y?: number, w?: number, h?: number): Chainable<string[][]>
+    table(
+      x?: number,
+      y?: number,
+      w?: number,
+      h?: number,
+    ): Chainable<string[][]>
 
     /**
      * Invokes the method on the current subject.

--- a/jsconfig.json
+++ b/jsconfig.json
@@ -1,6 +1,0 @@
-{
-  "compilerOptions": {
-    "lib": ["DOM", "ES6"],
-    "types": ["cypress"]
-  }
-}

--- a/package.json
+++ b/package.json
@@ -11,7 +11,8 @@
     "test": "cypress run",
     "badges": "npx -p dependency-version-badge update-badge cypress",
     "semantic-release": "semantic-release",
-    "stop-only": "stop-only --folder cypress/e2e"
+    "stop-only": "stop-only --folder cypress/e2e",
+    "lint": "tsc --noEmit --pretty"
   },
   "keywords": [
     "cypress-plugin"
@@ -23,7 +24,8 @@
     "cypress": "^12.5.1",
     "prettier": "^2.8.0",
     "semantic-release": "^20.0.0",
-    "stop-only": "^3.1.2"
+    "stop-only": "^3.1.2",
+    "typescript": "^4.9.5"
   },
   "peerDependencies": {
     "cypress": ">=12"

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,0 +1,10 @@
+{
+  "compilerOptions": {
+    "lib": ["DOM", "ES6"],
+    "types": ["cypress"],
+    "allowJs": true,
+    "noEmit": true,
+    "esModuleInterop": true
+  },
+  "include": ["commands/*.*", "cypress/e2e/**/*.js"]
+}

